### ISSUE TITLE
NRT in lang.interpret

### ISSUE THREAD

**Issue by VictorKolb** [crucialfelix/supercolliderjs#80]
Hello! Are there plans to implement NRT in lang.interpret? I try to do something like this:

```js
const sc = require("supercolliderjs");

sc.lang.boot({debug: true, echo: true, stdin: true, stdout: true}).then(async function (lang) {
    try {
        const res = await lang.interpret(`
SynthDef(\\pad, {
    arg freq = 440, dur = 1, amp = 1, feedback = 1, out = 0;
    var sig, local, env;
    sig = SinOsc.ar(freq);
    env = EnvGen.kr(
        Env.new([0, 1, 1, 0], [0, dur / 2, dur + 5], 'sin'),
        doneAction: 2
    );
    sig = sig * env;
    
    
    Out.ar(out, Pan2.ar(sig, SinOsc.kr([1, 2, 0.2, 0.5].choose, 0, [1, 0.5, 0.7].choose)));
}).store;


a = Scale.minorPentatonic;

p = Pbind(
    \\instrument, \\bass,
    \\ctranspose, 0,
    \\dur, Prand([1,2,4,8], inf),
    \\sustain, 1,
    \\degree, Pxrand(a.degrees, inf),
    \\out, 0
).asScore(duration: 8);


p.recordNRT(
    outputFilePath: "~/Desktop/111/test.wav".asAbsolutePath,
    headerFormat: "WAV",
    duration: 8
);

`);

        console.log(res)
    } catch (e) {
        console.log(e)
    }

    await lang.quit();
});

```

But i get test.wav with 44 bytes length.  :(

**Comment by crucialfelix** [comment]
Anything you run in sclang like that should be the exact same as running it from the SC IDE..

Do you need to boot the scsynth server? (I don't think so)
Do you have another scsynth server already running that conflicts with the one sclang is trying to use?

**Comment by crucialfelix** [comment]
btw.

```js
// ...
} catch (e) {
        console.log(e)
} finally {
    await lang.quit();
}
```

**Comment by VictorKolb** [comment]
@crucialfelix no one scsynth server is running. Same code in SC IDE will write file correct.

**Comment by crucialfelix** [comment]
Ah, it is because the sclang code finishes and returns (though it has
started the NRT process).

So the returned result (void) is returned to supercollider.js / node and it
then kills the lang.

There should be some way for users to message back to js land via
the SuperColliderJS class:

https://github.com/crucialfelix/supercolliderjs/blob/develop/packages/lang/src/supercollider-js/SuperColliderJS.sc

Or better yet, include a built in two way messaging system. Each side can
add their own event listeners..

Then you could call SuperColliderJS.send("DONE")

The node script needs to wait somehow, otherwise it will exit right away
because it's work seems to be done.

**Comment by VictorKolb** [comment]
@crucialfelix thank for answer! I'll try to do something with this)

**Comment by VictorKolb** [comment]
@crucialfelix hello again! After few attempt I not finding in documentation info about messaging system :( Or you mean is "need to create it"?

**Comment by crucialfelix** [comment]
Exactly. Paraphrased: Wouldn't it be a great idea if there was an easy way
to call messages back and forth?

PRs or proposals accepted of course.

**Comment by Catsvilles** [comment]
@VictorKolb Hey, did you had any success with this? I'm looking for a way to fully compose SC's score with JS

**Comment by VictorKolb** [comment]
@Catsvilles hey! I setting interval and every second looking at file size with `fs.statSync` if the file size is didn't changed for 3 times I conclude which rendering is has been done. This tactic good works for me.
